src/engine/engine.cpp

```diff
@@ -714,6 +715,15 @@ std::filesystem::path Engine::getSavegamePath(const std::optional<size_t>& slot)
     return root / QuicksaveFilename;
 }
 
+std::filesystem::path Engine::getSavegameMetaPath(const std::optional<size_t>& slot) const
+{
+  const auto root = getSavegameRootPath();
+  if(slot.has_value())
+    return root / makeSavegameMetaFilename(*slot);
+  else
+    return root / QuicksaveMetaFilename;
```

Comment:
See other comment in `World::save`. I'm also pretty sure this function becomes completely unnecessary with the proposed change.
```suggestion
  return makeMetaFilepath(getSavegamePath(slot));
```